pkg/aws/aws.go

```diff
@@ -42,24 +43,36 @@ const (
 	// Region represents AWS region
 	Region = "AWS_REGION"
 
+	// From AWS SDK "aws/session/env_config.go"
+	webIdentityTokenFilePathEnvKey = "AWS_WEB_IDENTITY_TOKEN_FILE"
+	roleARNEnvKey                  = "AWS_ROLE_ARN"
+
 	assumeRoleDuration = 25 * time.Minute
 )
 
-// GetConfigFromProfile extracts AWS creds from profile
-func GetConfigFromProfile(ctx context.Context, profile *param.Profile) (*aws.Config, string, error) {
-	config := make(map[string]string)
-
-	if profile.Credential.Type == param.CredentialTypeKeyPair {
-		config[AccessKeyID] = profile.Credential.KeyPair.ID
-		config[SecretAccessKey] = profile.Credential.KeyPair.Secret
-	} else if profile.Credential.Type == param.CredentialTypeSecret {
-		config[AccessKeyID] = string(profile.Credential.Secret.Data[secrets.AWSAccessKeyID])
-		config[SecretAccessKey] = string(profile.Credential.Secret.Data[secrets.AWSSecretAccessKey])
-		config[ConfigRole] = string(profile.Credential.Secret.Data[secrets.ConfigRole])
-		config[SessionToken] = string(profile.Credential.Secret.Data[secrets.AWSSessionToken])
+// GetCredentials returns credentials to use for AWS operations
+func GetCredentials(ctx context.Context, config map[string]string) (*credentials.Credentials, error) {
```

Comment:
Kanister packages now either call `GetCredentials` directly or they are using `ExtractAWSCredentials` (which calls `GetCredentials`).

The only exception is when the old-style `KeyPair` secret is being used. That will be deprecated going forward.